File: testers/avro/java-quarkus/generated-and-checked-in/com/example/events/OrderEventsListener.java
package com.example.events;

import io.smallrye.mutiny.Uni;
import org.eclipse.microprofile.reactive.messaging.Incoming;
import org.eclipse.microprofile.reactive.messaging.Message;
import org.eclipse.microprofile.reactive.messaging.Metadata;

/** Event listener interface for order-events topic. Implement this interface to handle events. */
public interface OrderEventsListener {
  /** Receive and dispatch events to handler methods */
  @Incoming("order-events")
  default Uni<Void> receive(Message<Object> record) {
    return switch (record.getPayload()) {
      case null -> onUnknown(record);
      case OrderCancelled e -> onOrderCancelled(e, record.getMetadata());
      case OrderPlaced e -> onOrderPlaced(e, record.getMetadata());
      case OrderUpdated e -> onOrderUpdated(e, record.getMetadata());
      default -> onUnknown(record);
    };
  }

  /** Handle OrderCancelled event */
  Uni<Void> onOrderCancelled(OrderCancelled event, Metadata metadata);

  /** Handle OrderPlaced event */
  Uni<Void> onOrderPlaced(OrderPlaced event, Metadata metadata);

  /** Handle OrderUpdated event */
  Uni<Void> onOrderUpdated(OrderUpdated event, Metadata metadata);

  /** Handle unknown event types. Override to customize behavior. */
  default Uni<Void> onUnknown(Message<Object> record) {
    return Uni.createFrom().voidItem();
  }
}

File: testers/avro/kotlin-quarkus-mutiny/generated-and-checked-in/com/example/events/OrderEventsListener.kt
package com.example.events

import io.smallrye.mutiny.Uni
import java.lang.Void
import org.eclipse.microprofile.reactive.messaging.Incoming
import org.eclipse.microprofile.reactive.messaging.Message
import org.eclipse.microprofile.reactive.messaging.Metadata

/** Event listener interface for order-events topic. Implement this interface to handle events. */
interface OrderEventsListener {
  /** Handle OrderCancelled event */
  abstract fun onOrderCancelled(
    event: OrderCancelled,
    metadata: Metadata
  ): Uni<Void>

  /** Handle OrderPlaced event */
  abstract fun onOrderPlaced(
    event: OrderPlaced,
    metadata: Metadata
  ): Uni<Void>

  /** Handle OrderUpdated event */
  abstract fun onOrderUpdated(
    event: OrderUpdated,
    metadata: Metadata
  ): Uni<Void>

  /** Handle unknown event types. Override to customize behavior. */
  fun onUnknown(record: Message<Any>): Uni<Void> {
    return Uni.createFrom().voidItem()
  }

  /** Receive and dispatch events to handler methods */
  @Incoming("order-events")
  fun receive(record: Message<Any>): Uni<Void> {
    return when (val __r = record.getPayload()) {
      null -> onUnknown(record)
      is OrderCancelled -> { val e = __r as OrderCancelled; onOrderCancelled(e, record.getMetadata()) }
      is OrderPlaced -> { val e = __r as OrderPlaced; onOrderPlaced(e, record.getMetadata()) }
      is OrderUpdated -> { val e = __r as OrderUpdated; onOrderUpdated(e, record.getMetadata()) }
      else -> onUnknown(record)
    }
  }
}